src/rapport/03_resultaten.Rmd

```diff
@@ -6,22 +6,142 @@ bibliography: moneos.bib
+```
+
+```{r eval=FALSE}
+# probleemgevallen met verschillende ecosysteem_werkelijk
+benthos_afdw_ecotoop %>%
+  filter(as.character(ecotoop_werkelijk.benthos) !=
+           as.character(ecotoop_werkelijk.ecotoop)) %>%
+  select(jaar, staal, starts_with("ecotoop"))
+```
+
+Enkel Zeeschelde, Rupel en Durme / geen raaien / enkel intertidaal
+
+Berekening stratum weight = 1/inclusiekans
 
-mzb_strata <- srv_data %>%
-  as_survey_design(strata = strata,
-                   weights = weights)
+Inclusiekans = stratum_n \* opp_per_staal_m2 / stratum_opp_m2
+
+```{r benthos-afdw-spatial}
+#diameter = 4.5 cm
+opp_per_staal_m2 <- pi * (4.5 / 100) ^ 2 / 4
+
+
+benthos_afdw_ecotoop %>%
+  select(jaar, survey, staal, staalcode, tot_afdw, tot_aantal,
+         ecotoop_werkelijk.benthos,
+         KRWzone,
+         SalZone,
+         Vallei_deel,
+         Naam,
+         stratum_opp_m2 = Shape_Area) %>%
+  filter(survey == "spatial",
+         Naam %in% c("Zeeschelde", "Durme"),
+         !str_detect(ecotoop_werkelijk.benthos, "subtidaal")) %>%
+  group_by(jaar, KRWzone, ecotoop_werkelijk.benthos) %>%
+  mutate(stratum_n = n(),
+         stratum_weight = stratum_opp_m2 / (opp_per_staal_m2 * stratum_n)
+         ) %>%
+  arrange(jaar, KRWzone, ecotoop_werkelijk.benthos, staal) %>%
+  ungroup() -> benthos_afdw_spatial
+```
+
+```{r schattingen}
+# data.frame maken met alle survey design variabele en responsvariabelen
+benthos_design <- benthos_afdw_spatial %>%
+  as_survey_design(strata = c(KRWzone, ecotoop_werkelijk.benthos),
+                   weights = stratum_weight)
 
 # https://cran.r-project.org/web/packages/srvyr/vignettes/srvyr-vs-survey.html
-# Horvitz-Thomson type schatters (eventueel eerst sqrt-transformeren;
-# of log, maar nullen)
-mzb_strata %>%
-  #group_by(...) %>%
-  summarize(mzb_tot = survey_total(x = responsvar, na.rm = TRUE,
-                                   vartype = "ci", level = 0.95))
+# Horvitz-Thomson type schatters
+benthos_design %>%
+  group_by(jaar) %>%
+  summarize(mzb_tonnage = survey_total(
+    x = tot_afdw / 1e6,
+    na.rm = TRUE,
+    vartype = "ci",
+    level = 0.95))
+
+# ofwel droge stof via zero-inflated lognormal distributie?
+# cf examples in ?survey::svymle (voor censored lognormal)
+# censored kan eventueel ook een optie zijn: alle nullen beschouwen als
+# links-gecensureerde data met kleine cutoff
+# zie ook ?survey::svysurvreg
+```
+
+```{r schattingen-subtargets}
+# data.frame maken met alle survey design variabele en responsvariabelen
+benthos_design <- benthos_afdw_spatial %>%
+  as_survey_design(strata = c(KRWzone, ecotoop_werkelijk.benthos),
+                   weights = stratum_weight)
 
+# https://cran.r-project.org/web/packages/srvyr/vignettes/srvyr-vs-survey.html
+# Horvitz-Thomson type schatters
+benthos_design %>%
+  group_by(jaar, SalZone) %>%
+  summarize(mzb_tonnage = survey_total(
+    x = tot_afdw / 1e6,
+    na.rm = TRUE,
+    vartype = "ci",
+    level = 0.95))
```

Comment:
Hier zijn er toch enkele CI waar 0 in ligt. Misschien is een censored / zero-inflated dan toch geen slecht idee?
Ongeveer 17% van de stalen heeft AFDW=0

Reply:
Voorlopig zou ik die piste even parkeren. Ik denk dat we best voorrang geven aan de vraag over raaien / spatial afstemming.

Reply:
Ik heb gisteren nog veel gezocht maar lognormaal gaat volgens mij al niet simpel zijn! Er zijn enkele voorbeelden te vinden maar die zijn meestal niet heel gedetailleerd uitgewerkt.